### DISCUSSION TITLE
Add command to toggle inline diagnostics

### DISCUSSION THREAD

**Discussion by thomasschafer** [helix-editor/helix#11230] [General]
I'm really enjoying the new inline diagnostics that was recently merged into master, and I'm so grateful for the work that went in to this.

Sometimes I find it a little distracting to have diagnostics appear and disappear when I'm editing, shifting around the lines below my cursor. I think it would be great to have a new command to show inline diagnostics below the line, which would then disappear either via a command or when the cursor is moved to a new line. This could be paired with `end-of-line-diagnostics` being enabled and `editor.inline-diagnostics.cursor-line` disabled, so that diagnostics still show up at the end of the line, but only below the line when requested.

**Comment by pascalkuthe** [comment]
> Sometimes I find it a little distracting to have diagnostics appear and disappear when I'm editing, shifting around the lines below my cursor.

that's why inline diagnostics are disabled in insert-mode and only appear after a timeout when moving the cursor.

> disabled, so that diagnostics still show up at the end of the line, but only below the line when requested.

you can use `:toggle` to change the `inline-diagnostics.cursor-line` setting to achieve this. Simply toggle inline diagnostics on when you want to see them and off when you don't want to see them. I don't think adding a dedicated command beyond that is worth it

**Reply by thomasschafer** [reply]
I guess so, but I still think that being able to toggle it temporarily (like with the `hover` command mapped to `space + k`) is preferable.

Good point about the `:toggle` workaround, ~~although I've noticed that there appears to be a bug there - I expected `:toggle inline-diagnostics.cursor-line warning` to work, but that shows an error, presumably because [this line](https://github.com/helix-editor/helix/blob/748a9cf022eb74e96a3697e4b11b2490b1e58f08/helix-term/src/commands/typed.rs#L1956) checks `> 2` rather than `>= 2`. I guess that's because of the cycling ability added [here](https://github.com/helix-editor/helix/pull/4411), but I think using `>=2` should resolve this issue without breaking cycling. I've created a PR to fix that [here](https://github.com/helix-editor/helix/pull/11232).~~

Edit: ignore the above - for posterity, the command I wanted is either `:set inline-diagnostics.cursor-line warning` to enable it, or `:toggle inline-diagnostics.cursor-line warning disable` to flip it on and off.

**Reply by arunoruto** [reply]
Just came upon this thread and the `:toggle` solution helped a lot!
I made some quick keybindings, so I don't have to type it out ;)
```toml
[keys.normal.space.i]
c = ":toggle inline-diagnostics.cursor-line hint disable"
e = ":toggle end-of-line-diagnostics warning disable"
o = ":toggle inline-diagnostics.other-lines error disable"
```
Note: `d` and `D` were already taken for the diagnostic panel for the project and workspace, respectively. So I just picked `i` for `inline`. Maybe this helps someone :)

**Comment by nik-rev** [comment]
If/when https://github.com/helix-editor/helix/pull/12203 gets merged you'll be able to do it
